pkg/serviceaccounts/oauthclient/oauthclientregistry_test.go

```diff
@@ -577,8 +607,14 @@ func TestGetClient(t *testing.T) {
 			t.Errorf("%s: expected %#v, got %#v", tc.name, tc.expectedOSActions, tc.routeClient.Actions())
 			continue
 		}
-	}
 
+		if len(tc.expectedEventMsg) > 0 {
+			ev := <-fakerecorder.Events
```

Comment:
I think you want a select statement with a default to prevent this from ever dead locking.